### PR TITLE
ITS for Spider

### PR DESCRIPTION
**Pull request by andr2000**

### CODE REVIEW

File: xen/common/xmalloc_tlsf.c [hunk 1]
```diff
@@ -548,6 +549,17 @@ static void *xmalloc_whole_pages(unsigned long size, unsigned long align)
     return res;
 }
 
+void *_xzalloc_whole_pages(unsigned long size, unsigned long align,
```
Comment (lorc) [inline xen/common/xmalloc_tlsf.c:552]: Introduction of an `_xzalloc_whole_pages` should go in a separate patch.

File: xen/arch/arm/gic-v3-its.c [hunk 2]
```diff
@@ -61,6 +61,76 @@ struct its_device {
...
+{
+    if ( its_quirk_mask & HOST_ITS_WORKAROUND_R8A779F0 )
+        return GIC_BASER_NonShareable;
+
+    return GIC_BASER_InnerShareable;
+}
+
+unsigned int gicv3_its_get_memflags(void)
+{
+    if ( its_quirk_mask & HOST_ITS_WORKAROUND_R8A779F0 )
```
Comment (lorc) [inline xen/arch/arm/gic-v3-its.c]: I'm not sure that we need one `HOST_ITS_WORKAROUND_R8A779F0` flag to rule them all.

We need to check: are there other ITS implementation that support only 32 bit addresses? Maybe we should have flags like `HOST_ITS_WORKAROUND_32BIT_ADDRESSES` and `HOST_ITS_WORKAROUND_NC_NS` at least.

File: xen/arch/arm/gic-v3-its.c [hunk 3]
```diff
@@ -61,6 +61,76 @@ struct its_device {
...
+ * so assume they all share the same.
+ */
+struct its_quirk {
+    const char *desc;
+    bool (*init)(struct host_its *hw_its);
+    uint32_t iidr;
+    uint32_t mask;
+};
+
+uint32_t its_quirk_mask;
```
Comment (andr2000) [inline xen/arch/arm/gic-v3-its.c]: flags

File: xen/arch/arm/gic-v3-its.c (hunk 2 above)
Comment (andr2000) [reply xen/arch/arm/gic-v3-its.c]: agree

File: xen/common/xmalloc_tlsf.c (hunk 1 above)
Comment (andr2000) [reply xen/common/xmalloc_tlsf.c:552]: will keep it integrated for now until we see this is needed

**Comment by lorc** [comment]
`Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`
